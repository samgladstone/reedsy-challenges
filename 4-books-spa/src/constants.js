const BOOK_SORTS = {
    author: 'Author',
    published: 'Published',
    rating: 'Rating',
    title: 'Title',
}

export {
    BOOK_SORTS
}