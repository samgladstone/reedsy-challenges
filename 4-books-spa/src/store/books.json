[
  {
    "id": 1,
    "author": "Markus Zusak",
    "cover": "11.jpg",
    "rating": "9.9",
    "published": "2005",
    "slug": "the-book-theif",
    "synopsis": "1939. Nazi Germany. The country is holding its breath. Death has never been busier. Liesel, a nine-year-old girl, is living with a foster family on Himmel Street. Her parents have been taken away to a concentration camp. Liesel steals books. This is her story and the story of the inhabitants of her street when the bombs begin to fall.",
    "title": "The Book Thief",
    "upvoted": false,
    "upvotes": 1264,
    "stores": {
      "amazon": "https://www.amazon.co.uk/Book-Thief-Markus-Zusak/dp/0552773891",
      "iBooks": "https://books.apple.com/gb/book/the-book-theif/id1078821147",
      "playStore": "https://play.google.com/store/audiobooks/details/Markus_Zusak_The_Book_Thief?id=AQAAAEDgxmqGmM&hl=en"
    }
  },
  {
    "id": 2,
    "author": "David Eagleman",
    "cover": "12.jpg",
    "rating": "9.8",
    "published": "2009",
    "slug": "sum-forty-tales-from-the-afterlives",
    "synopsis": "In the afterlife you may find that God is the size of a microbe and unaware of your existence. Or you may find the afterlife contains only those people whom you remember. In some afterlives you are split into all your different ages; in some you are recreated based on your credit-card records; and in others you are forced to live with annoying versions of yourself that represent what you could have been. In these wonderfully imagined tales - at once funny, wistful and unsettling - Eagleman kicks over the chessboard of traditional notions and offers us a dazzling lens through which to see ourselves here and now. His stories are rooted in science and romance and awe at our mysterious existence: a mixture of hope, love and death that cuts through human nature at innovative angles. ",
    "title": "Sum: Forty Tales from the Afterlives",
    "upvoted": false,
    "upvotes": 1388,
    "stores": {
      "iBooks": "https://books.apple.com/gb/book/sum/id388400637",
      "playStore": "https://play.google.com/store/books/details/David_Eagleman_Sum?id=0TYdCyyir2IC&hl=en"
    }
  },
  {
    "id": 3,
    "author": "Matt Haig",
    "cover": "13.jpg",
    "rating": "9.7",
    "published": "2013",
    "slug": "the-humans",
    "synopsis": "THERE'S NO PLACE LIKE HOME. OR IS THERE? After an 'incident' one wet Friday night where he is found walking naked through the streets of Cambridge, Professor Andrew Martin is not feeling quite himself. Food sickens him. Clothes confound him. Even his loving wife and teenage son are repulsive to him. He feels lost amongst an alien species and hates everyone on the planet. Everyone, that is, except Newton, and he's a dog. Who is he really? And what could make someone change their mind about.",
    "title": "The Humans",
    "upvoted": false,
    "upvotes": 2424,
    "stores": {
      "amazon": "https://www.amazon.co.uk/Humans-Matt-Haig/dp/1786894661",
      "playStore": "https://play.google.com/store/books/details/Matt_Haig_The_Humans?id=4mZ1XoSpR9UC&hl=en"
    }
  },
  {
    "id": 4,
    "author": "Iain M. Banks",
    "cover": "14.jpg",
    "rating": "9.6",
    "published": "2012",
    "slug": "the-hydrogen-sonata",
    "synopsis": "The Scavenger species are circling. It is, truly, the End Days for the Gzilt civilisation.\n\nAn ancient people, organised on military principles and yet almost perversely peaceful, the Gzilt helped set up the Culture ten thousand years earlier and were very nearly one of its founding societies, deciding not to join only at the last moment. Now they've made the collective decision to follow the well-trodden path of millions of other civilisations: they are going to Sublime, elevating themselves to a new and almost infinitely more rich and complex existence.\n\nAmid preparations though, the Regimental High Command is destroyed. Lieutenant Commander (reserve) Vyr Cossont appears to have been involved, and she is now wanted - dead, not alive. Aided only by an ancient, reconditioned android and a suspicious Culture avatar, Cossont must complete her last mission given to her by the High Command. She must find the oldest person in the Culture, a man over nine thousand years old, who might have some idea what really happened all that time ago.\n\nIt seems that the final days of the Gzilt civilisation are likely to prove its most perilous.",
    "title": "The Hydrogen Sonata",
    "upvoted": false,
    "upvotes": 2424,
    "stores": {
      "amazon": "https://www.amazon.co.uk/Hydrogen-Sonata-Culture-Iain-Banks/dp/0356501493",
      "iBooks": "https://books.apple.com/gb/book/the-hydrogen-sonata/id543917639"
    }
  },
  {
    "id": 5,
    "author": "Patrick Rothfuss",
    "cover": "15.jpg",
    "rating": "9.5",
    "published": "2010",
    "slug": "the-name-of-the-wind",
    "synopsis": "I have stolen princesses back from sleeping barrow kings. I burned down the town of Trebon. I have spent the night with Felurian and left with both my sanity and my life. I was expelled from the University at a younger age than most people are allowed in. I tread paths by moonlight that others fear to speak of during day. I have talked to Gods, loved women, and written songs that make the minstrels weep. My name is Kvothe. You may have heard of me.\n\nSo begins the tale of Kvothe - currently known as Kote, the unassuming innkeeper - from his childhood in a troupe of traveling players, through his years spent as a near-feral orphan in a crime-riddled city, to his daringly brazen yet successful bid to enter a difficult and dangerous school of magic.\n\nIn these pages you will come to know Kvothe the notorious magician, the accomplished thief, the masterful musician, the dragon-slayer, the legend-hunter, the lover, the thief and the infamous assassin. The Name of the Wind is fantasy at its very best, and an astounding must-read title.",
    "title": "The Name of the Wind",
    "upvoted": false,
    "upvotes": 1745,
    "stores": {
      "playStore": "https://play.google.com/store/books/details/Patrick_Rothfuss_The_Name_of_the_Wind?id=BcG2dVRXKukC&hl=en"
    }
  },
  {
    "id": 6,
    "author": "Marcel Proust",
    "cover": "01.jpg",
    "rating": "8.9",
    "published": "1913",
    "slug": "in-search-of-lost-time",
    "synopsis": "In Search of Lost Time, also translated as Remembrance of Things Past, novel in seven parts by Marcel Proust, published in French as À la recherche du temps perdu from 1913 to 1927. The novel is the story of Proust's own life, told as an allegorical search for truth.\nIn Search of Lost Time, also translated as Remembrance of Things Past, novel in seven parts by Marcel Proust, published in French as À la recherche du temps perdu from 1913 to 1927. The novel is the story of Proust's own life, told as an allegorical search for truth.",
    "title": "In Search of Lost Time",
    "upvoted": false,
    "upvotes": 1111,
    "stores": {
      "amazon": "https://www.amazon.co.uk/Search-Lost-Time-Penguin-Classics/dp/0141180315"
    }
  },
  {
    "id": 7,
    "author": "Miguel de Cervantes",
    "cover": "02.jpg",
    "rating": "8.8",
    "published": "1605",
    "slug": "don-quixote",
    "synopsis": "Don Quixote is a middle-aged gentleman from the region of La Mancha in central Spain. Obsessed with the chivalrous ideals touted in books he has read, he decides to take up his lance and sword to defend the helpless and destroy the wicked.\nDon Quixote is a middle-aged gentleman from the region of La Mancha in central Spain. Obsessed with the chivalrous ideals touted in books he has read, he decides to take up his lance and sword to defend the helpless and destroy the wicked.",
    "title": "Don Quixote",
    "upvoted": true,
    "upvotes": 1022,
    "stores": {
      "iBooks": "https://books.apple.com/gb/book/don-quixote/id1190143157"
    }
  },
  {
    "id": 8,
    "author": "James Joyce",
    "cover": "03.jpg",
    "rating": "8.6",
    "published": "1922",
    "slug": "ulysses",
    "synopsis": "All the action of Ulysses takes place in and immediately around Dublin on a single day (June 16, 1904). The three central characters—Stephen Dedalus (the hero of Joyce’s earlier Portrait of the Artist as a Young Man); Leopold Bloom, a Jewish advertising canvasser; and his wife, Molly—are intended to be modern counterparts of Telemachus, Ulysses (Odysseus), and Penelope, respectively, and the events of the novel loosely parallel the major events in Odysseus’s journey home after the Trojan War.\nAll the action of Ulysses takes place in and immediately around Dublin on a single day (June 16, 1904). The three central characters—Stephen Dedalus (the hero of Joyce’s earlier Portrait of the Artist as a Young Man); Leopold Bloom, a Jewish advertising canvasser; and his wife, Molly—are intended to be modern counterparts of Telemachus, Ulysses (Odysseus), and Penelope, respectively, and the events of the novel loosely parallel the major events in Odysseus’s journey home after the Trojan War.",
    "title": "Ulysses",
    "upvoted": false,
    "upvotes": 1003
  },
  {
    "id": 9,
    "author": "F. Scott Fitzgerald",
    "cover": "04.jpg",
    "rating": "8.3",
    "published": "1925",
    "slug": "the-great-gatsby",
    "synopsis": "The Great Gatsby is a story told by Nick Carraway, who was once Gatsby's neighbor, and he tells the story sometime after 1922, when the incidents that fill the book take place. As the story opens, Nick has just moved from the Midwest to West Egg, Long Island, seeking his fortune as a bond salesman.\nThe Great Gatsby is a story told by Nick Carraway, who was once Gatsby's neighbor, and he tells the story sometime after 1922, when the incidents that fill the book take place. As the story opens, Nick has just moved from the Midwest to West Egg, Long Island, seeking his fortune as a bond salesman.",
    "title": "The Great Gatsby",
    "upvoted": false,
    "upvotes": 991,
    "stores": {
      "amazon": "https://www.amazon.co.uk/Great-Gatsby-Wordsworth-Classics/dp/185326041X"
    }
  },
  {
    "id": 10,
    "author": "Herman Melville",
    "cover": "05.jpg",
    "rating": "8.1",
    "published": "1851",
    "slug": "moby-dick",
    "synopsis": "The sole survivor of a lost whaling ship relates the tale of his captain's self-destructive obsession to hunt the white whale, Moby Dick. This classic story by Herman Melville revolves around Captain Ahab and his obsession with a huge whale, Moby Dick.\nThe sole survivor of a lost whaling ship relates the tale of his captain's self-destructive obsession to hunt the white whale, Moby Dick. This classic story by Herman Melville revolves around Captain Ahab and his obsession with a huge whale, Moby Dick.",
    "title": "Moby Dick",
    "upvoted": true,
    "upvotes": 940,
    "stores": {
      "iBooks": "https://books.apple.com/gb/book/moby-dick/id427802472",
      "playStore": "https://play.google.com/store/books/details/Herman_Melville_Moby_Dick?id=XV8XAAAAYAAJ&hl=en"
    }
  },
  {
    "id": 11,
    "author": "William Shakespeare",
    "cover": "06.jpg",
    "rating": "8.0",
    "published": "1609",
    "slug": "hamlet",
    "synopsis": "Hamlet, Prince of Denmark, is home from school to mourn the death of his father, King Hamlet, who has died two months earlier. Hamlet is disgusted by the marriage of his newly widowed mother, Queen Gertrude, to his Uncle, King Hamlet's brother, Claudius, who now has the throne.\nHamlet, Prince of Denmark, is home from school to mourn the death of his father, King Hamlet, who has died two months earlier. Hamlet is disgusted by the marriage of his newly widowed mother, Queen Gertrude, to his Uncle, King Hamlet's brother, Claudius, who now has the throne.",
    "title": "Hamlet",
    "upvoted": false,
    "upvotes": 909,
    "stores": {
      "amazon": "https://www.amazon.co.uk/Hamlet-Oxford-Shakespeare-Worlds-Classics/dp/0199535817",
      "playStore": "https://play.google.com/store/books/details/William_Shakespeare_Hamlet?id=BA9GAAAAcAAJ&hl=en"
    }
  },
  {
    "id": 12,
    "author": "Leo Tolstoy",
    "cover": "07.jpg",
    "rating": "7.8",
    "published": "1865",
    "slug": "war-and-peace",
    "synopsis": "War and Peace broadly focuses on Napoleon’s invasion of Russia in 1812 and follows three of the most well-known characters in literature: Pierre Bezukhov, the illegitimate son of a count who is fighting for his inheritance and yearning for spiritual fulfillment; Prince Andrei Bolkonsky, who leaves his family behind to fight in the war against Napoleon; and Natasha Rostov, the beautiful young daughter of a nobleman who intrigues both men.\nWar and Peace broadly focuses on Napoleon’s invasion of Russia in 1812 and follows three of the most well-known characters in literature: Pierre Bezukhov, the illegitimate son of a count who is fighting for his inheritance and yearning for spiritual fulfillment; Prince Andrei Bolkonsky, who leaves his family behind to fight in the war against Napoleon; and Natasha Rostov, the beautiful young daughter of a nobleman who intrigues both men.",
    "title": "War and Peace",
    "upvoted": true,
    "upvotes": 839,
    "stores": {
      "amazon": "https://www.amazon.co.uk/War-Peace-Penguin-Clothbound-Classics/dp/0241265541"
    }
  },
  {
    "id": 13,
    "author": "Homer",
    "cover": "08.jpg",
    "rating": "7.5",
    "published": "800 B.C.E",
    "slug": "the-odyssey",
    "synopsis": "With the consent of Zeus, Athena travels to Ithaca to speak with Telemachus. Assuming the form of Odysseus's old friend Mentes, Athena predicts that Odysseus is still alive and that he will soon return to Ithaca. She advises Telemachus to call together the suitors and announce their banishment from his father's estate.\nWith the consent of Zeus, Athena travels to Ithaca to speak with Telemachus. Assuming the form of Odysseus's old friend Mentes, Athena predicts that Odysseus is still alive and that he will soon return to Ithaca. She advises Telemachus to call together the suitors and announce their banishment from his father's estate.",
    "title": "The Odyssey",
    "upvoted": false,
    "upvotes": 802,
    "stores": {
      "amazon": "https://www.amazon.co.uk/Odyssey-Penguin-Classics-Homer/dp/0140449116",
      "playStore": "https://play.google.com/store/books/details/Homer_The_Odyssey?id=zNOmwC1aodEC&hl=en"
    }
  },
  {
    "id": 14,
    "author": "Gabriel Garcia Marquez",
    "cover": "09.jpg",
    "rating": "7.2",
    "published": "1967",
    "slug": "one-hundred-years-of-solitude",
    "synopsis": "The patriarch of the family, José Arcadio Buendía, is impulsive and inquisitive. He remains a leader who is also deeply solitary, alienating himself from other men in his obsessive investigations into mysterious matters. These character traits are inherited by his descendents throughout the novel.\nThe patriarch of the family, José Arcadio Buendía, is impulsive and inquisitive. He remains a leader who is also deeply solitary, alienating himself from other men in his obsessive investigations into mysterious matters. These character traits are inherited by his descendents throughout the novel.",
    "title": "One Hundred Years of Solitude",
    "upvoted": true,
    "upvotes": 723,
    "stores": {
      "playStore": "https://play.google.com/store/books/details/Gabriel_Garcia_Marquez_One_Hundred_Years_of_Solitu?id=RyGVAgAAQBAJ&hl=en"
    }
  },
  {
    "id": 15,
    "author": "Dante Alighieri",
    "cover": "10.jpg",
    "rating": "7.1",
    "published": "1472",
    "slug": "the-divine-comedy",
    "synopsis": "In the Divine comedy poem, he is the main character. The plot is set up in 1300, around the time Dante was in exile from his native Florence. But the actual writing was done by him between 1308 and 1321 at the time of his death. The Divine Comedy summary gives a picture of the realms of Dante's afterlife.\nIn the Divine comedy poem, he is the main character. The plot is set up in 1300, around the time Dante was in exile from his native Florence. But the actual writing was done by him between 1308 and 1321 at the time of his death. The Divine Comedy summary gives a picture of the realms of Dante's afterlife.",
    "title": "The Divine Comedy",
    "upvoted": false,
    "upvotes": 749,
    "stores": {
      "amazon": "https://www.amazon.co.uk/Divine-Comedy-Purgatorio-Paradiso-Classics/dp/0141197498"
    }
  }
]