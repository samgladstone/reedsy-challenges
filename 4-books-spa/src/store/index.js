import Vue from 'vue';
import Vuex from 'vuex';
import books from './books.json';
import { sortByTextProperty, sortByTextYearProperty } from '../helpers';

Vue.use(Vuex);

const PAGE_SIZE = 5;
const calculateTotalPages = (itemCount = 0) => Math.ceil(itemCount / PAGE_SIZE)

export const getters = {
    booksSorted: (state, getters) => Object.values(state.books).sort(getters.sortFn(state.sortBy, getters.sortDesc)),
    page: (state, getters) => (
        start => getters.booksSorted.slice(start, start + PAGE_SIZE)
    )((state.pageNumber - 1) * PAGE_SIZE),
    pagesTotal: (s, getters) => calculateTotalPages(getters.booksSorted.length),
    sortFn: state => state.sortBy === 'published' ? sortByTextYearProperty : sortByTextProperty,
    /** Switch it around, but also account for rating being odd in that higher numbers should come first  */
    sortDesc: state => state.sortBy === 'rating' ? state.sortAsc : !state.sortAsc,
}

export const mutations = {
    changePage(state, { pageNumber }) {
        if (!pageNumber && pageNumber !== 0)
            throw new Error('A new page number is required.');
        if (!Number.isInteger(pageNumber))
            throw new Error('The page number must be an integer.');
        if (pageNumber < 1)
            throw new Error('Cannot go below page 1.');
        if (pageNumber > calculateTotalPages(Object.keys(state.books).length))
            throw new Error('Cannot go beyond the last page.');

        state.pageNumber = pageNumber;
    },
    changeSort(state, { sortAsc, sortBy }) {
        state.sortAsc = sortAsc === !!sortAsc ? sortAsc : state.sortAsc;
        state.sortBy = sortBy || state.sortBy;
    },
};

let id = 1;

export default new Vuex.Store({
    strict: process.env.NODE_ENV !== 'production',
    state: {
        books: books.reduce((all, book) => ({ ...all, [id]: { id: id++, ...book } }), {}),
        pageNumber: 1,
        sortAsc: true,
        sortBy: 'rating',
    },
    getters,
    mutations,
});