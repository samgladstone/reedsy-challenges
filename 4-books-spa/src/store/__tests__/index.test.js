import store, { mutations, getters } from '../index';
import { BOOK_SORTS } from '../../constants';

describe('Getters', () => {
    describe('booksSorted', () => {
        let mockBooksSorted = (sortAsc, sortBy) => getters.booksSorted({
            books: {
                1: {
                    "id": 1,
                    "author": "Markus Zusak",
                    "rating": "9.9",
                    "published": "2005",
                    "title": "The Book Thief",
                },
                2: {
                    "id": 2,
                    "author": "David Eagleman",
                    "rating": "9.8",
                    "published": "2009",
                    "title": "Sum: Forty Tales from the Afterlives",

                },
                3: {
                    "id": 3,
                    "author": "Matt Haig",
                    "rating": "9.7",
                    "published": "2013",
                    "title": "The Humans",
                }
            },
            sortAsc,
            sortBy,
        });

        test('Default sort settings', () => {
            expect(store.state.sortAsc).toBe(false);
            expect(store.state.sortBy).toBe('rating');
        });

        Object.keys(BOOK_SORTS).forEach(sort => {
            describe(`Sort by ${sort}`, () => {
                test('Ascending', () => { expect(mockBooksSorted(true, sort)).toMatchSnapshot(); });
                test('Descending', () => { expect(mockBooksSorted(false, sort)).toMatchSnapshot(); });
            });
        });
    });
});

describe('Mutations', () => {
    describe('changeSort', () => {
        let changeSort, mockState;

        beforeAll(() => { changeSort = mutations.changeSort });

        beforeEach(() => {
            mockState = {
                sortAsc: false,
                sortBy: 'rating',
            }
        });

        test('The sort order can be changed', () => {
            changeSort(mockState, { sortAsc: true });
            expect(mockState.sortAsc).toBe(true);
            expect(mockState.sortBy).toBe('rating');
        });

        test('The sort by can be changed', () => {
            changeSort(mockState, { sortBy: 'author' });
            expect(mockState.sortAsc).toBe(false);
            expect(mockState.sortBy).toBe('author');
        });

        test('Sort by and direction can be updated together', () => {
            changeSort(mockState, { sortAsc: true, sortBy: 'author' });
            expect(mockState.sortAsc).toBe(true);
            expect(mockState.sortBy).toBe('author');
        });

        test('Not passing in a sort by or order preserves the original state', () => {
            changeSort(mockState, {});
            expect(mockState.sortAsc).toBe(false);
            expect(mockState.sortBy).toBe('rating');
        });
    });
});