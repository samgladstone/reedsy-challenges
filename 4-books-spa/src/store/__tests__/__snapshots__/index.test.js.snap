// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Getters booksSorted Sort by author Ascending 1`] = `
Array [
  Object {
    "author": "David Eagleman",
    "id": 2,
    "published": "2009",
    "rating": "9.8",
    "title": "Sum: Forty Tales from the Afterlives",
  },
  Object {
    "author": "Markus Zusak",
    "id": 1,
    "published": "2005",
    "rating": "9.9",
    "title": "The Book Thief",
  },
  Object {
    "author": "Matt Haig",
    "id": 3,
    "published": "2013",
    "rating": "9.7",
    "title": "The Humans",
  },
]
`;

exports[`Getters booksSorted Sort by author Descending 1`] = `
Array [
  Object {
    "author": "Matt Haig",
    "id": 3,
    "published": "2013",
    "rating": "9.7",
    "title": "The Humans",
  },
  Object {
    "author": "Markus Zusak",
    "id": 1,
    "published": "2005",
    "rating": "9.9",
    "title": "The Book Thief",
  },
  Object {
    "author": "David Eagleman",
    "id": 2,
    "published": "2009",
    "rating": "9.8",
    "title": "Sum: Forty Tales from the Afterlives",
  },
]
`;

exports[`Getters booksSorted Sort by published Ascending 1`] = `
Array [
  Object {
    "author": "Markus Zusak",
    "id": 1,
    "published": "2005",
    "rating": "9.9",
    "title": "The Book Thief",
  },
  Object {
    "author": "David Eagleman",
    "id": 2,
    "published": "2009",
    "rating": "9.8",
    "title": "Sum: Forty Tales from the Afterlives",
  },
  Object {
    "author": "Matt Haig",
    "id": 3,
    "published": "2013",
    "rating": "9.7",
    "title": "The Humans",
  },
]
`;

exports[`Getters booksSorted Sort by published Descending 1`] = `
Array [
  Object {
    "author": "Matt Haig",
    "id": 3,
    "published": "2013",
    "rating": "9.7",
    "title": "The Humans",
  },
  Object {
    "author": "David Eagleman",
    "id": 2,
    "published": "2009",
    "rating": "9.8",
    "title": "Sum: Forty Tales from the Afterlives",
  },
  Object {
    "author": "Markus Zusak",
    "id": 1,
    "published": "2005",
    "rating": "9.9",
    "title": "The Book Thief",
  },
]
`;

exports[`Getters booksSorted Sort by rating Ascending 1`] = `
Array [
  Object {
    "author": "Matt Haig",
    "id": 3,
    "published": "2013",
    "rating": "9.7",
    "title": "The Humans",
  },
  Object {
    "author": "David Eagleman",
    "id": 2,
    "published": "2009",
    "rating": "9.8",
    "title": "Sum: Forty Tales from the Afterlives",
  },
  Object {
    "author": "Markus Zusak",
    "id": 1,
    "published": "2005",
    "rating": "9.9",
    "title": "The Book Thief",
  },
]
`;

exports[`Getters booksSorted Sort by rating Descending 1`] = `
Array [
  Object {
    "author": "Markus Zusak",
    "id": 1,
    "published": "2005",
    "rating": "9.9",
    "title": "The Book Thief",
  },
  Object {
    "author": "David Eagleman",
    "id": 2,
    "published": "2009",
    "rating": "9.8",
    "title": "Sum: Forty Tales from the Afterlives",
  },
  Object {
    "author": "Matt Haig",
    "id": 3,
    "published": "2013",
    "rating": "9.7",
    "title": "The Humans",
  },
]
`;

exports[`Getters booksSorted Sort by title Ascending 1`] = `
Array [
  Object {
    "author": "David Eagleman",
    "id": 2,
    "published": "2009",
    "rating": "9.8",
    "title": "Sum: Forty Tales from the Afterlives",
  },
  Object {
    "author": "Markus Zusak",
    "id": 1,
    "published": "2005",
    "rating": "9.9",
    "title": "The Book Thief",
  },
  Object {
    "author": "Matt Haig",
    "id": 3,
    "published": "2013",
    "rating": "9.7",
    "title": "The Humans",
  },
]
`;

exports[`Getters booksSorted Sort by title Descending 1`] = `
Array [
  Object {
    "author": "Matt Haig",
    "id": 3,
    "published": "2013",
    "rating": "9.7",
    "title": "The Humans",
  },
  Object {
    "author": "Markus Zusak",
    "id": 1,
    "published": "2005",
    "rating": "9.9",
    "title": "The Book Thief",
  },
  Object {
    "author": "David Eagleman",
    "id": 2,
    "published": "2009",
    "rating": "9.8",
    "title": "Sum: Forty Tales from the Afterlives",
  },
]
`;
